cypress/integration/tests/controls/stakeholders/pagination.test.ts

```diff
@@ -0,0 +1,169 @@
+                        const stakeholder = new Stakeholders(data.getEmail(), data.getFullName());
+                        stakeholder.create();
+                        stakeholdersList.push(stakeholder);
+                    }
+                }
+            });
+    });
+
+    beforeEach("Persist session", function () {
+        // Save the session and token cookie for maintaining one login session
+        Cypress.Cookies.preserveOnce("AUTH_SESSION_ID", "KEYCLOAK_SESSION");
+
+        // Interceptors
+        cy.intercept("GET", "/api/controls/stakeholder*").as("getStakeholders");
+    });
+
+    after("Perform test data clean up", function () {
+        // Delete the stakeholders created before the tests
+        if (stakeholdersList.length > 0) {
+            stakeholdersList.forEach(function (stakeholder) {
+                stakeholder.delete();
+            });
+        }
+    });
+
+    it("Navigation button validations", function () {
+        // Navigate to stakeholder tab
+        clickByText(navMenu, controls);
+        clickByText(navTab, stakeholders);
+        cy.wait("@getStakeholders");
+
+        // select 10 items per page
+        selectItemsPerPage(10);
+        cy.wait("@getStakeholders");
+
+        // Verify next buttons are enabled as there are more than 11 rows present
+        cy.get(nextPageButton).each(($nextBtn) => {
+            cy.wrap($nextBtn).should("not.be.disabled");
+        });
+
+        // Verify that previous buttons are disabled being on the first page
+        cy.get(prevPageButton).each(($previousBtn) => {
+            cy.wrap($previousBtn).should("be.disabled");
+        });
+
+        // Verify that navigation button to last page is enabled
+        cy.get(lastPageButton).should("not.be.disabled");
+
+        // Verify that navigation button to first page is disabled being on the first page
+        cy.get(firstPageButton).should("be.disabled");
+
+        // Navigate to next page
+        cy.get(nextPageButton).eq(0).click();
+        cy.wait("@getStakeholders");
+
+        // Verify that previous buttons are enabled after moving to next page
+        cy.get(prevPageButton).each(($previousBtn) => {
+            cy.wrap($previousBtn).should("not.be.disabled");
+        });
+
+        // Verify that navigation button to first page is enabled after moving to next page
+        cy.get(firstPageButton).should("not.be.disabled");
+    });
+
+    it("Items per page validations", function () {
+        // Navigate to stakeholder tab
+        clickByText(navMenu, controls);
+        clickByText(navTab, stakeholders);
+        cy.wait("@getStakeholders");
+
+        // Select 10 items per page
+        selectItemsPerPage(10);
+        cy.wait(2000);
+
+        // Verify that only 10 items are displayed
+        cy.get("td[data-label=Email]").then(($rows) => {
+            var rowCount = $rows.length;
+            cy.wrap($rows.length).should("eq", 10);
```

Comment:
Did you miss using that var rowCount or it has any other reason?
  `cy.wrap(rowCount).should("eq", 10);`

Reply:
Right, I missed that var. Good catch !